### PR TITLE
UCT/IB/UD: Fix multi-path handling for passive endpoint

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Fix multi-path handling for passive endpoint.

## Why ?

If RoCE LAG feature is used to spread traffic across two paths, address held in passive endpoint is incorrect, because it always use `path_index = 0`.
The issue is `uct_ud_ep_create_passive` function should set `path_index` prior invoking `uct_ep_connect_to_ep` function which uses `path_index` to unpack UD EP address.

## How ?

1. Introduce `uct_ud_ep_create` function which fills `uct_ep_params`, invokes `uct_ep_create` function, sets `conn_sn`.
2. Use `uct_ud_ep_create` in `uct_ud_ep_create_passive` and `uct_ud_ep_create_connected_common` to reduce duplication and fix bug.

### CODE REVIEW

**Comment by dmitrygx** [comment]
the fix is setting `path_index` before invoking `uct_ep_connect_to_ep` during creating a passive endpoint

**Comment by dmitrygx** [comment]
@brminich @hoopoepg could you review pls?

**Review by brminich — DISMISSED** [review]

**Comment by dmitrygx** [comment]
@yosefe could you review pls?

**Review by hoopoepg — APPROVED** [review]

File: src/uct/ib/ud/base/ud_ep.c [hunk 1]
```diff
@@ -569,18 +591,12 @@ ucs_status_t uct_ud_ep_create_connected_common(const uct_ep_params_t *ep_params,
...
-
-    status = uct_ep_create(&params, &new_ep_h);
+    status = uct_ud_ep_create(iface, path_index, conn_sn, &ep);
     if (status != UCS_OK) {
         goto out;
     }
 
-    ep          = ucs_derived_of(new_ep_h, uct_ud_ep_t);
-    ep->conn_sn = conn_sn;
+    *new_ep_p = &ep->super.super;
```
Comment (yosefe) [inline src/uct/ib/ud/base/ud_ep.c:599]: already in line 633

Comment (dmitrygx) [reply src/uct/ib/ud/base/ud_ep.c]: removed, and force-pushed (PR is small enough)

**Review by yosefe — APPROVED** [review]

**Pull request merged**